scenarios/matRad_ImportanceScenarios.m

```diff
@@ -0,0 +1,236 @@
+        name = 'impScen';
+    end
+
+    properties (Constant)
+        validCombinationTypes = {'all','none','shift'};
+    end
+    
+    methods
+        function this = matRad_ImportanceScenarios(ct)           
+            if nargin == 0 
+                superclassArgs = {};
+            else
+                superclassArgs = {ct};
+            end
+            
+            this@matRad_ScenarioModel(superclassArgs{:});
+
+            %TODO: We could do this automatically in the superclass
+            %Octave 5 has a bug there and throws an error
+            this.updateScenarios();
+        end
+
+        function set.combineRange(this,combineRange_)
+            valid = isscalar(combineRange_) && (isnumeric(combineRange_) || islogical(combineRange_));
+            if ~valid 
+                matRad_cfg = MatRad_Config.instance();
+                matRad_cfg.dispError('Invalid value for combineRange! Needs to be a boolean / logical value!');
+            end
+            this.combineRange = combineRange_;
+            this.updateScenarios();
+        end
+        
+        function scenarios = updateScenarios(this)
+            matRad_cfg = MatRad_Config.instance();
+              
+            %Get the maximum, i.e., worst case shifts
+            wcSetupShifts = this.wcSigma * this.shiftSD;
+            
+            %% Create gridded setup shifts
+            %Create grid vectors for setup shifts
+            setupShiftGrid = zeros(this.numOfSetupGridPoints,numel(wcSetupShifts));            
+            if mod(this.numOfSetupGridPoints,2) == 0
+                matRad_cfg.dispWarning('Obtaining Setup Shifts: Including the nominal scenario with even number of grid points creates asymmetrical shifts!');
+            end
+
+            for i = 1:numel(wcSetupShifts)
+                setupShiftGrid(:,i) = linspace(-wcSetupShifts(i),wcSetupShifts(i),this.numOfSetupGridPoints);
+                if this.includeNominalScenario 
+                      
+                    [~,ix] = min(abs(setupShiftGrid(:,i)));
+                    setupShiftGrid(ix,i) = 0;    
+                end  
+            end
+            
+            %Now create vector of all shifts for different combinatorial
+            %settings
+            switch this.combinations
+                case 'none'
+                    %Create independent shifts
+                    griddedSetupShifts = [];
+                    for i=1:size(setupShiftGrid,2)
+                        tmpGrid = zeros(size(setupShiftGrid,1),3);
+                        tmpGrid(:,i) = setupShiftGrid(:,i);
+                        griddedSetupShifts = [griddedSetupShifts; tmpGrid];
+                    end                    
+                case 'shift'
+                    [X,Y,Z] = meshgrid(setupShiftGrid(:,1),setupShiftGrid(:,2),setupShiftGrid(:,3));
+                    griddedSetupShifts = [X(:), Y(:), Z(:)];    
+                case 'all'
+                    error('Not implemented!');
```

Comment:
what do you mean by the "all" combinations setup ?

Reply:
basically every combination of range & shift scenarios is modeled. shift means only shift combinations are modeled, with range errors separate. none means every dimension/error is just modeled individually.
I thought though that I had implemented the "all" combinations.. Need to check..